internal/topology/variables/cluster_variable_defaulting_test.go

```diff
@@ -220,20 +220,206 @@ func Test_DefaultClusterVariables(t *testing.T) {
 				},
 			},
 		},
+
+		// Note this test should be flaky if it fails as it is testing the ordering of the elements in the array.
```

Comment:
I added a large number of variables and mixed up the ordering here so this test should fail at least sometimes if this code doesn't work. Is there any better way to test completely that the order is always right?

Reply:
You could make it a separate test outside of this table which:
* generates a clusterClassVariables array with way more variables and/or
* runs the Defaulting with the same data repeatedly

But I would implement neither of those options in this table.

Reply:
Both of these would still have the same issue of being flaky rather than failing, right? I'd like a way to deterministically fail if possible. It seems this test fails often enough - at least on my local set up - to know when the ordering is not happening correctly.

Reply:
Imho there's no way to get it to fail every time. Afaik go intentionally randomizes the order (so that nobody relies on it) so there is also no guarantee that it will fail as you can always get "lucky" :P

Reply:
I'm fine with this test coverage for now, and with the consideration that if the implementation is changed back to a non-determininstic order, things can still pass by chance (but I assume that with 10 variables it will fail most of the time).
Eventually, in the future, we can create a dedicated test and use the fuzzer to generate a huge number of combinations, but I don't think it is required at this stage 
```suggestion
		// If the implementation changes and the deterministic order is dropped for any reason, there is
		// a chance the test will pass when the random order matches the desired one; thus generating flakes.
		// Eventually in future we can consider to hardening this test using the fuzzer or increasing the number of variables 
```